dashboard/app/models/user.rb

```diff
@@ -1084,14 +1084,16 @@ def next_unpassed_visible_progression_level(script)
 
     last_visible_level = visible_sls.max_by(&:chapter)
 
+    # If the user started but didn't finish a level, go to that level.
+    # Or if the user completed the last level but not all previous levels.
+    return most_recent_sl if most_recent_sl == last_visible_level || !most_recent_ul.passing?
+
     visible_incomplete_sls = visible_sls.find_all {|sl| visible_incomplete_level_ids.include?(sl.level_id)}
 
     first_visible_incomplete_level = visible_incomplete_sls.min_by(&:chapter)
 
-    # The user has completed the last level in the progression, but not all
-    # previous levels, return the first visible incomplete script_level
     return first_visible_incomplete_level || first_visible_level if
-      (most_recent_sl == last_visible_level) || most_recent_sl.nil?
+      most_recent_sl.nil?
```

Comment:
I'm surprised this check isn't up several lines.  In fact, if the user has no progress in the script, I think we'd get a nil reference error on line 1003 above.  I'm not sure why we don't in the test case "can get next_unpassed_visible_progression_level, no progress, none hidden."